dwave_networkx/algorithms/partitioning.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2021 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from dwave_networkx import chimera_graph
+from dimod import DiscreteQuadraticModel
+from dwave.system import LeapHybridDQMSampler
+import networkx as nx
+import numpy as np
+
+__all__ = ["partition", "weighted_partition"]
+
+
+def partition(G, num_partitions=2, lagrange=4, sampler=None, **sampler_args):
```

Comment:
Docstring for `sampler` mentions using `set_default_sampler` when no sampler is provided.  I think you'll need to add a decorator to support that.
```suggestion
@binary_quadratic_model_sampler(3)
def partition(G, num_partitions=2, lagrange=4, sampler=None, **sampler_args):
```

Reply:
I removed the decorator because the dqm sampler, LeapHybridDQMSampler, is not validated by that decorator and I missed that in the docstring. Is there an equivalent of @binary_quadratic_model_sampler for dqm?

Reply:
Good point, it doesn't look like there is a decorator for DQM.  I would just remove that part of the docstring for the `sampler` arg.